diff --git a/node_modules/@types/react-table/index.d.ts b/node_modules/@types/react-table/index.d.ts
index 636829c..38990f4 100644
--- a/node_modules/@types/react-table/index.d.ts
+++ b/node_modules/@types/react-table/index.d.ts
@@ -282,7 +282,7 @@ export interface UseTableHeaderGroupProps<D extends object> {
 
 export interface UseTableColumnProps<D extends object> {
     id: IdType<D>;
-    columns: Array<ColumnInstance<D>>;
+    columns?: Array<ColumnInstance<D>>;
     isVisible: boolean;
     render: (type: 'Header' | 'Footer' | string, props?: object) => ReactNode;
     totalLeft: number;
@@ -290,10 +290,9 @@ export interface UseTableColumnProps<D extends object> {
     getHeaderProps: (propGetter?: HeaderPropGetter<D>) => TableHeaderProps;
     getFooterProps: (propGetter?: FooterPropGetter<D>) => TableFooterProps;
     toggleHidden: (value?: boolean) => void;
-    parent: ColumnInstance<D>; // not documented
+    parent?: ColumnInstance<D>; // not documented
     getToggleHiddenProps: (userProps?: any) => any;
     depth: number; // not documented
-    index: number; // not documented
     placeholderOf?: ColumnInstance;
 }
 
