diff --git a/node_modules/react-scripts/config/webpack.config.js b/node_modules/react-scripts/config/webpack.config.js
index d34d06f..1f94181 100644
--- a/node_modules/react-scripts/config/webpack.config.js
+++ b/node_modules/react-scripts/config/webpack.config.js
@@ -338,6 +338,9 @@ module.exports = function(webpackEnv) {
         // Disable require.ensure as it's not a standard language feature.
         { parser: { requireEnsure: false } },
 
+        // dump test file that were mistakenly shipped as part of graphql-language-service
+        { test: /\.spec\.js$/, use: 'null-loader' },
+
         // First, run the linter.
         // It's important to do this before Babel processes the JS.
         {
