diff --git a/dist/index.js b/dist/index.js
index 1a0cccf..ae2a698 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -34,26 +34,21 @@ function _possibleConstructorReturn(self, call) { if (!self) { throw new Referen

 function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } // eslint-disable-line import/no-extraneous-dependencies

-
-if (!process.env.REACT_SPINKIT_NO_STYLES) {
-  /* eslint-disable global-require */
-  require('loaders.css');
-  require('../css/base.css');
-  require('../css/loaders-css.css');
-  require('../css/fade-in.css');
-  require('../css/chasing-dots.css');
-  require('../css/circle.css');
-  require('../css/cube-grid.css');
-  require('../css/double-bounce.css');
-  require('../css/folding-cube.css');
-  require('../css/pulse.css');
-  require('../css/rotating-plane.css');
-  require('../css/three-bounce.css');
-  require('../css/wandering-cubes.css');
-  require('../css/wave.css');
-  require('../css/wordpress.css');
-  /* eslint-enable global-require */
-}
+import 'loaders.css';
+import '../css/base.css';
+import '../css/loaders-css.css';
+import '../css/fade-in.css';
+import '../css/chasing-dots.css';
+import '../css/circle.css';
+import '../css/cube-grid.css';
+import '../css/double-bounce.css';
+import '../css/folding-cube.css';
+import '../css/pulse.css';
+import '../css/rotating-plane.css';
+import '../css/three-bounce.css';
+import '../css/wandering-cubes.css';
+import '../css/wave.css';
+import '../css/wordpress.css';

 var noFadeInWarning = "Deprecation Warning (react-spinkit): noFadeIn prop should be replaced with fadeIn='none'";

@@ -126,4 +121,4 @@ Spinner.defaultProps = {
   overrideSpinnerClassName: ''
 };

-module.exports = Spinner;
\ No newline at end of file
+export default Spinner;
diff --git a/src/index.jsx b/src/index.jsx
index e18c479..954196e 100644
--- a/src/index.jsx
+++ b/src/index.jsx
@@ -1,6 +1,6 @@
 import PropTypes from 'prop-types';
 import React from 'react'; // eslint-disable-line import/no-extraneous-dependencies
-import cx from 'classnames';
+import cx from 'clsx';
 import assign from 'object-assign';
 import { allSpinners } from './spinners';

