diff --git a/config/webpack.config.js b/config/webpack.config.js
index e465d8e7a00ec889e9c4c29ba7812019f8319a41..65348115c40b18f41bdf968862d97ab67762509c 100644
--- a/config/webpack.config.js
+++ b/config/webpack.config.js
@@ -584,6 +584,10 @@ module.exports = function (webpackEnv) {
                 'sass-loader'
               ),
             },
+            {
+              test: /\.mdx?$/,
+              use: ['babel-loader', '@mdx-js/loader', path.join(__dirname, '../../../../../../support/webpack/fm-loader')]
+            },
             // "file" loader makes sure those assets get served by WebpackDevServer.
             // When you `import` an asset, you get its (virtual) filename.
             // In production, they would get copied to the `build` folder.
