diff --git a/dist/ts-internals.js b/dist/ts-internals.js
index 2b303c90edf91dcc3bf9988586570b9988f05187..6040e67f820c58e3a4ee110b5fadc4bcc2f2e832 100644
--- a/dist/ts-internals.js
+++ b/dist/ts-internals.js
@@ -37,6 +37,7 @@ function createTsInternalsUncached(_ts) {
         const resolved = ts.nodeModuleNameResolver(extendedConfig, combinePaths(basePath, 'tsconfig.json'), { moduleResolution: ts.ModuleResolutionKind.NodeJs }, host, 
         /*cache*/ undefined, 
         /*projectRefs*/ undefined, 
+        /*conditionsOrIsConfigLookup*/ undefined,
         /*lookupConfig*/ true);
         if (resolved.resolvedModule) {
             return resolved.resolvedModule.resolvedFileName;
